BotBrownCore/Configuration/GreetingConfiguration.cs

```diff
@@ -3,15 +3,15 @@
     using System.Collections.Generic;
     using System.ComponentModel;
 
-    public class GreetingConfiguration : IConfiguration
+    public class GreetingConfiguration : IChangeableConfiguration
     {
-        public Dictionary<string, string> Greetings { get; set; } = new Dictionary<string, string>();
-
+        public Dictionary<string, string> Greetings { get; set; }
```

Comment:
Why did you change this from an initialized Dictionary? I don't see a constructor.